db.go

```diff
@@ -660,6 +666,31 @@ func (a *dbAppender) Rollback() error {
 	return g.Wait()
 }
 
+// Delete implements deletion of metrics.
+func (db *DB) Delete(mint, maxt int64, ms ...labels.Matcher) error {
+	db.cmtx.Lock()
+	defer db.cmtx.Unlock()
+
+	db.headmtx.RLock()
```

Comment:
A compaction can also only change non-head blocks.
In the querier we hold the entire DB lock regardless, which we don't do here, right? Not 100% sure without looking properly again why we lock headmtx for the brief retrieval of `blocksForRange` – probably there was a reason somewhere